src/main/java/doore/study/application/CurriculumItemCommandService.java

```diff
@@ -29,54 +33,105 @@ public class CurriculumItemCommandService {
                         .build())
                 .toList();
         participantCurriculumItemRepository.saveAll(participantCurriculumItems);
     }
 
-    public void deleteCurriculum(Long curriculumId) {
-        CurriculumItem curriculumItem = curriculumItemRepository.findById(curriculumId)
-                .orElseThrow(() -> new CurriculumItemException(NOT_FOUND_CURRICULUM_ITEM));
+    private void updateCurriculum(List<CurriculumItem> curriculumItems) {
+        for (CurriculumItem requestItem : curriculumItems) {
+            CurriculumItem existingItem = curriculumItemRepository.findById(requestItem.getId())
+                    .orElseThrow(() -> new CurriculumItemException(NOT_FOUND_CURRICULUM_ITEM));
 
-        int deletedOrder = curriculumItem.getItemOrder();
-        curriculumItemRepository.delete(curriculumItem);
+            if (!existingItem.getName().equals(requestItem.getName())) {
+                existingItem.updateName(requestItem.getName());
+            }
 
-        List<CurriculumItem> subsequentItems = curriculumItemRepository.findByItemOrderGreaterThan(deletedOrder);
-        for (CurriculumItem item : subsequentItems) {
-            item.updateOrder(item.getItemOrder() - 1);
+            if (!existingItem.getItemOrder().equals(requestItem.getItemOrder())) {
+                existingItem.updateItemOrder(requestItem.getItemOrder());
+            }
         }
     }
 
-    public void updateCurriculum(Long curriculumId, CurriculumItemRequest request) {
-        CurriculumItem curriculumItem = curriculumItemRepository.findById(curriculumId)
-                .orElseThrow(() -> new CurriculumItemException(NOT_FOUND_CURRICULUM_ITEM));
-        curriculumItem.update(request.name());
+    private void deleteCurriculum(List<CurriculumItem> deletedCurriculumItems) {
+        for (CurriculumItem requestItem : deletedCurriculumItems) {
+            curriculumItemRepository.deleteById(requestItem.getId());
+        }
     }
 
-    public void checkCurriculum(Long curriculumId) {
-        CurriculumItem curriculumItem = curriculumItemRepository.findById(curriculumId)
-                .orElseThrow(() -> new CurriculumItemException(NOT_FOUND_CURRICULUM_ITEM));
-        ParticipantCurriculumItem participantCurriculumItem = participantCurriculumItemRepository.findById(
-                curriculumItem.getId()).orElseThrow(); //todo: 예외처리
-        if (participantCurriculumItem.getIsChecked().equals(false)) participantCurriculumItem.complete();
-        else participantCurriculumItem.incomplete();
-    }
+    private void sortCurriculum() {
+        List<CurriculumItem> sortedCurriculum = curriculumItemRepository.findAll()
+                .stream()
+                .sorted(Comparator.comparingInt(CurriculumItem::getItemOrder))
+                .toList();
 
+        for (int i = 0; i < sortedCurriculum.size(); i++) {
+            CurriculumItem item = sortedCurriculum.get(i);
+            item.updateItemOrder(i + 1);
+        }
```

Comment:
```suggestion
         IntStream.range(0, sortedCurriculum.size())
                .forEach(i -> sortedCurriculum.get(i).updateItemOrder(i + 1));
```
이렇게 작성할 수 있겠네요~ Nit입니다~! `IntStream` 가독성이 별로인 것 같다면 반영 안하셔도 되요!